src/components/Carousel.ts

```diff
@@ -121,37 +137,87 @@ export default defineComponent({
       Object.assign(config, newConfig)
     }
 
-    const handleResize = debounce(() => {
+    const handleResize = throttle(() => {
       updateBreakpointsConfig()
       updateSlidesData()
       updateSlideSize()
     })
 
+    function parseTransform(el: HTMLElement) {
+      const transform = window.getComputedStyle(el).transform
+
+      //add sanity check
+      return transform
+        .split(/[(,)]/)
+        .slice(1, -1)
+        .map(function (v) {
+          return parseFloat(v)
+        })
+    }
+
+    const totalGap = computed(() => (config.itemsToShow - 1) * config.gap)
+    const transformElements = shallowReactive<Array<HTMLElement>>([])
+
     /**
      * Setup functions
      */
     function updateSlideSize(): void {
       if (!viewport.value) return
-      const rect = viewport.value.getBoundingClientRect()
-
-      // Calculate the total gap space
-      const totalGap = (config.itemsToShow - 1) * config.gap
+      let multiplierWidth = 1
+      let multiplierHeight = 1
+      transformElements.forEach((el) => {
+        const transformArr = parseTransform(el)
+
+        if (transformArr.length == 6) {
+          multiplierWidth *= transformArr[0]
+          multiplierHeight *= transformArr[3]
+        }
+      })
 
       // Calculate size based on orientation
       if (isVertical.value) {
-        slideSize.value = (rect.height - totalGap) / config.itemsToShow
+        const track = viewport.value.querySelector('.carousel__track')
+        if (config.height !== 'auto') {
+          let height
+          if (typeof config.height === 'string') {
+            const fakeEl = document.createElement('div')
+            fakeEl.style.height = trackHeight.value!
+            track?.append(fakeEl)
+            height = fakeEl.getBoundingClientRect().height
+            track?.removeChild(fakeEl)
+          } else {
+            height = config.height
+          }
+
+          slideSize.value =
+            (height / multiplierWidth - totalGap.value) / config.itemsToShow
+        } else if (slides.length) {
+          const height =
+            slides.reduce((total, slide) => {
+              const el = slide.vnode.el
+              if (!el) {
+                return total
+              }
+              const cloned = el.cloneNode(true)
+              cloned.style.width = null
+              cloned.style.height = 'auto'
+              track?.append(cloned)
+
+              const height = cloned.getBoundingClientRect().height
+
+              track?.removeChild(cloned)
+              return total + height
+            }, 0) / slides.length
+          slideSize.value = height / multiplierHeight
+        }
```

Comment:
This addition introduces significant complexity to the component. For now, let's keep it simple by relying on a fixed height. We can introduce this feature in a separate PR later as this PR is already quite large.